src/lib.rs

```diff
@@ -152,16 +160,56 @@ macro_rules! impl_argminmax {
 }
 
 // Implement ArgMinMax for the rust primitive types
-impl_nb_bits!(false, i8 i16 i32 i64 u8 u16 u32 u64);
-impl_nb_bits!(true, f32 f64);
 impl_argminmax!(i8, i16, i32, i64, f32, f64, u8, u16, u32, u64);
-
 // Implement ArgMinMax for other data types
 #[cfg(feature = "half")]
-mod half_impl {
+impl_argminmax!(f16);
+
+// ------------------------------ Vec ------------------------------
+
+macro_rules! impl_argminmax_vec {
+    ($($t:ty),*) => {
+        $(
+            impl ArgMinMax for Vec<$t> {
+                fn argminmax(self) -> (usize, usize) {
+                    self.as_slice().argminmax()
+                }
+            }
+        )*
+    };
+}
+
+impl_argminmax_vec!(i8, i16, i32, i64, f32, f64, u8, u16, u32, u64);
+#[cfg(feature = "half")]
+impl_argminmax_vec!(f16);
+
+// ----------------------- (optional) ndarray ----------------------
+
+#[cfg(feature = "ndarray")]
+mod ndarray_impl {
     use super::*;
-    use half::f16;
+    use ndarray::{Array1, ArrayView1};
+
+    // Use the slice implementation
+    // -> implement for T where slice implementation available
+    macro_rules! impl_argminmax_ndarray {
+        ($($t:ty),*) => {
+            $(
+                impl ArgMinMax for ArrayView1<'_, $t> {
+                    fn argminmax(self) -> (usize, usize) {
+                        self.as_slice().unwrap().argminmax()
+                    }
+                }
+                impl ArgMinMax for Array1<$t> {
+                    fn argminmax(self) -> (usize, usize) {
+                        self.as_slice().unwrap().argminmax()
+                    }
+                }
```

Comment:
Not really sure, but I guess we can implement the trait for 1D `ArrayBase` (which should cover both `ArrayView1` and `Array1`